File: backend/dtos/src/main/java/io/mateu/dtos/Action.java
package io.mateu.dtos;

public record Action(
    String id,
    String caption,
    ActionType type,
    boolean visible,
    boolean validationRequired,
    boolean confirmationRequired,
    boolean rowsSelectedRequired,
    ConfirmationTexts confirmationTexts,
    ActionTarget target,
    String modalStyle,
    String customEvent,
    String href) {}

File: frontend/web/vaadin/src/mateu/shared/apiClients/dtos/Action.ts
import {ActionType} from "./ActionType";
import ConfirmationTexts from "./ConfirmationTexts";
import {ActionTarget} from "./ActionTarget";
import {ActionPosition} from "./ActionPosition";

export default interface Action {

    id: string
    icon: string
    caption: string
    type: ActionType
    variants: string[]
    visible: boolean
    validationRequired: boolean
    confirmationRequired: boolean
    rowsSelectedRequired: boolean
    confirmationTexts: ConfirmationTexts
    target: ActionTarget
    modalStyle: string
    modalTitle: string
    customEvent: string
    href: string
    runOnEnter: boolean
    position: ActionPosition
    timeoutMillis: number

}
